internal/agent/device/bootstrap.go

```diff
@@ -110,6 +120,68 @@ func (b *Bootstrap) Initialize(ctx context.Context) error {
 	return nil
 }
 
+func (b *Bootstrap) EnsureCurrentRenderedSpecUpToDate(ctx context.Context) error {
+	currentSpec, err := spec.ReadRenderedSpecFromFile(b.deviceReader, b.currentRenderedFile)
+	if err != nil {
+		// During the initial bootstrap it is expected that the spec does not exist yet.  This assumption is validated later.
+		if errors.Is(err, spec.ErrMissingRenderedSpec) {
+			return nil
+		}
+		return fmt.Errorf("getting current rendered spec: %w", err)
+	}
+
+	desiredSpec, err := spec.ReadRenderedSpecFromFile(b.deviceReader, b.desiredRenderedFile)
+	if err != nil {
+		// During the initial bootstrap it is expected that the spec does not exist yet.  This assumption is validated later.
+		if errors.Is(err, spec.ErrMissingRenderedSpec) {
+			return nil
+		}
+		return fmt.Errorf("getting desired rendered spec: %w", err)
+	}
+
+	if !isOsImageInTransition(&currentSpec, &desiredSpec) {
+		// We didn't change the OS image, so nothing to do here
+		return nil
+	}
+
+	bootcCmd := container.NewBootcCmd(b.executer)
+	bootcHost, err := bootcCmd.Status(ctx)
+	if err != nil {
+		return fmt.Errorf("getting current bootc status: %w", err)
+	}
+
+	if container.IsOsImageReconciled(bootcHost, &desiredSpec) {
+		err = spec.WriteRenderedSpecToFile(b.deviceWriter, &desiredSpec, b.currentRenderedFile)
+		if err != nil {
+			return fmt.Errorf("writing rendered spec to file: %w", err)
+		}
+	} else {
+		// We rebooted without applying the new OS image - something went wrong
+		b.log.Warnf("%sstarted bootstrap with OS image not equal to desired image", b.logPrefix)
+		condErr := fmt.Errorf("booted image %s, expected %s", container.GetImage(bootcHost), desiredSpec.Os.Image)
+		err = b.statusManager.UpdateConditionError(ctx, BootedWithUnexpectedImage, condErr)
+		if err != nil {
+			b.log.Warnf("%sfailed setting error condition: %v", b.logPrefix, err)
+			return fmt.Errorf("setting error condition: %w", err)
+		}
+		return fmt.Errorf("os image was not reconciled")
```

Comment:
would this fail bootstrap? we just need `Warn` via conditions/log but it is a possible condition that can be reconciled.

Reply:
done